allauth/account/views.py

```diff
@@ -293,6 +293,9 @@ def get(self, *args, **kwargs):
             self.object = self.get_object()
             if app_settings.CONFIRM_EMAIL_ON_GET:
                 return self.post(*args, **kwargs)
+            email = self.object.email_address.email
+            if EmailAddress.objects.filter(email=email, verified=True).exists() and app_settings.EMAIL_CONFIRMATION_UNVERIFIED_ONLY is True:
```

Comment:
Doing this check here results in this not working when `CONFIRM_EMAIL_ON_GET` is `True`.

Wouldn't `EmailConfirmationHMAC::from_key()` be a better place? The `EmailAddress` lookup is already there.

Reply:
Ok good catch I am fine with that.